lib/ansible/module_utils/redfish_utils.py

```diff
@@ -0,0 +1,932 @@
+        else:
+            return {'ret': False, 'msg': 'Invalid Command'}
+
+        response = self.post_request(self.root_uri + action_uri, payload, HEADERS)
+        if response['ret'] is False:
+            return response
+        result['ret'] = True
+        return result
+
+    def list_users(self):
+        # listing all users has always been slower than other operations, why?
+        result = {}
+        allusers = []
+        allusers_details = []
+        response = self.get_request(self.root_uri + self.accounts_uri)
+        if response['ret'] is False:
+            return response
+        data = response['data']
+
+        for users in data[u'Members']:
+            allusers.append(users[u'@odata.id'])   # allusers[] are URIs
+
+        # for each user, get details
+        for uri in allusers:
+            response = self.get_request(self.root_uri + uri)
+            if response['ret'] is False:
+                return response
+            data = response['data']
+
+            if not data[u'UserName'] == "":        # only care if name is not empty
+                user = {}
+                user['Id'] = data[u'Id']
+                user['Name'] = data[u'Name']
+                user['UserName'] = data[u'UserName']
+                user['RoleId'] = data[u'RoleId']
+                allusers_details.append(user)
+        return {'ret': True, 'entries': allusers_details}
+
+    def add_user(self, user):
+        result = {}
+        uri = self.root_uri + self.accounts_uri + "/" + user['userid']
+        username = {'UserName': user['username']}
+        pswd = {'Password': user['userpswd']}
+        roleid = {'RoleId': user['userrole']}
+        enabled = {'Enabled': True}
+        for payload in username, pswd, roleid, enabled:
+            response = self.patch_request(uri, payload, HEADERS)
+            if response['ret'] is False:
+                return response
+        return {'ret': True}
+
+    def enable_user(self, user):
+        result = {}
+        uri = self.root_uri + self.accounts_uri + "/" + user['userid']
+        payload = {'Enabled': True}
+        response = self.patch_request(uri, payload, HEADERS)
+        if response['ret'] is False:
+            return response
+        return {'ret': True}
+
+    def delete_user(self, user):
+        result = {}
+        uri = self.root_uri + self.accounts_uri + "/" + user['userid']
+        payload = {'UserName': ""}
+        response = self.patch_request(uri, payload, HEADERS)
+        if response['ret'] is False:
+            return response
+        return {'ret': True}
+
+    def disable_user(self, user):
+        result = {}
+        uri = self.root_uri + self.accounts_uri + "/" + user['userid']
+        payload = {'Enabled': False}
+        response = self.patch_request(uri, payload, HEADERS)
+        if response['ret'] is False:
+            return response
+        return {'ret': True}
+
+    def update_user_role(self, user):
+        result = {}
```

Comment:
Should this be `response` since that is what is returned? I don't see `result` used within this function.